Tasks/Common/MSBuildHelpers/ArgumentFunctions.ps1

```diff
@@ -20,8 +20,39 @@ function Format-MSBuildArguments {
             $MSBuildArguments = "$MSBuildArguments /p:VisualStudioVersion=`"$VSVersion`""
         }
 
+        $userAgent = Get-UserAgentString
+        if (!([string]::IsNullOrEmpty($userAgent))) {
+            $MSBuildArguments = "$MSBuildArguments /p:_MSDeployUserAgent=`"$userAgent`""
```

Comment:
It is applicable when we deploy to azure from msbuild/vsbuild.
No nothing in user agent needs to be escaped